frontend/components/GamesVisibilityList.tsx

```diff
@@ -2,40 +2,46 @@ import * as React from 'react';
 import { connect } from 'react-redux';
 import { Dispatch, AnyAction } from 'redux';
 import { AppStore, GameVisibilityActions } from '../storage/common';
-import { CheckboxRowProps, CheckboxRow } from './generics/CheckboxRow';
+import { CheckboxRow } from './generics/CheckboxRow';
+import { Game } from '../storage/models/Game';
 interface GamesVisibilityListProps {
-	checkboxRows: Array<CheckboxRowProps>;
+	gamesVisibility: Map<Game, boolean>;
 	dispatch: Dispatch<AnyAction>;
 }
+
 const GamesVisibilityList: React.FC<GamesVisibilityListProps> = (props: GamesVisibilityListProps) => {
-	const callbackFunction = (name: string, checkValue: boolean) => {
-		const gameNumber = parseInt(name);
+	const callbackFunction = (value: string, checkValue: boolean) => {
+		const gameNumber = parseInt(value);
 		props.dispatch(GameVisibilityActions.setGameVisibility(gameNumber, !checkValue));
 	};
-	const { checkboxRows } = props;
+	const { gamesVisibility } = props;
+	const arrayOfGames = Array.from(gamesVisibility.entries());
```

Comment:
Okay, at the moment what you do is:
 1. Create a new data structure to have a dictionary (I'll be using dictionary when referring to the `Map()` class to avoid confusion with `Array.map()`) of game -> visibility
 2. Convert that structure to an array of entries
 3. Iterate over the newly created array.

----

An alternative solution would be:
 1. Pass the array of games and the map of gameId -> Visibility
 2. Iterate over the games array

Specifically the code would then look like this:
```
// map state to props
return {
    games: Array.from(state.games),
    visibilityMap: state.gamesVisibility
};

// Renderer
const callbackFunction = ...;
const {games, visibilityMap} = props;
return (
    <>
        {games.map(game => {
            const isGameVisible = visibilityMap.get(game.id) ?? true;

            return (
                <CheckboxRow
                    key={game.id}
                    id={`gameVisibility_${game.id}`}
                    label={game.name}
                    defaultCheckValue={isGameVisible}
                    value={`${game.id}`}
                    callback={callbackFunction}
                />
            );
        })}
    </>
)
```

----

Is it a better solution? I could argue both ways. You slightly flatten the structure at the cost of pushing some logic to the renderer. You could also just improve your code by extracting the game and visibility from entry to avoid the hard-to-parse `entry[0]`. I'll leave it up to you to make the decision whether to keep your way or change it to the one I presented.
But if you keep it the way it is just please destructrure `entry` first for better clarity in the map callback.